app/controllers/v0/sessions_controller.rb

```diff
@@ -73,10 +78,14 @@ def parse_current_loa
       LOA::MAPPING[raw_loa]
     end
 
-    def create_new_user
-      user = User.new(user_attributes)
-      user = Decorators::MviUserDecorator.new(user).create unless user.loa1?
-      user
+    def saml_user
+      @user || create_saml_user
```

Comment:
What's the difference between `@user` and `@current_user`? Does `@user` need to be an instance variable; could it just be a return value?